examples/http_server.rb

```diff
@@ -0,0 +1,169 @@
+require "json"
+require "logger"
+
+# Create a simple tool
+class ExampleTool < MCP::Tool
+  description "A simple example tool that adds two numbers"
+  input_schema(
+    properties: {
+      a: { type: "number" },
+      b: { type: "number" },
+    },
+    required: ["a", "b"],
+  )
+
+  class << self
+    def call(a:, b:)
+      MCP::Tool::Response.new([{
+        type: "text",
+        text: "The sum of #{a} and #{b} is #{a + b}",
+      }])
+    end
+  end
+end
+
+# Create a simple prompt
+class ExamplePrompt < MCP::Prompt
+  description "A simple example prompt that echoes back its arguments"
+  arguments [
+    MCP::Prompt::Argument.new(
+      name: "message",
+      description: "The message to echo back",
+      required: true,
+    ),
+  ]
+
+  class << self
+    def template(args, server_context:)
+      MCP::Prompt::Result.new(
+        messages: [
+          MCP::Prompt::Message.new(
+            role: "user",
+            content: MCP::Content::Text.new(args[:message]),
+          ),
+        ],
+      )
+    end
+  end
+end
+
+# Set up the server
+server = MCP::Server.new(
+  name: "example_http_server",
+  tools: [ExampleTool],
+  prompts: [ExamplePrompt],
+  resources: [
+    MCP::Resource.new(
+      uri: "test_resource",
+      name: "Test resource",
+      description: "Test resource that echoes back the uri as its content",
+      mime_type: "text/plain",
+    ),
+  ],
+)
+
+server.define_tool(
+  name: "echo",
+  description: "A simple example tool that echoes back its arguments",
+  input_schema: { properties: { message: { type: "string" } }, required: ["message"] },
+) do |message:|
+  MCP::Tool::Response.new(
+    [
+      {
+        type: "text",
+        text: "Hello from echo tool! Message: #{message}",
+      },
+    ],
+  )
+end
+
+server.resources_read_handler do |params|
+  [{
+    uri: params[:uri],
+    mimeType: "text/plain",
+    text: "Hello from HTTP server resource!",
+  }]
+end
+
+# Create the Streamable HTTP transport
+transport = MCP::Server::Transports::StreamableHTTPTransport.new(server)
+server.transport = transport
+
+# Create a logger for MCP-specific logging
+mcp_logger = Logger.new($stdout)
+mcp_logger.formatter = proc do |_severity, _datetime, _progname, msg|
+  "[MCP] #{msg}\n"
+end
+
+# Create a Rack application with logging
+app = proc do |env|
+  request = Rack::Request.new(env)
+
+  # Log MCP-specific details for POST requests
+  if request.post?
+    body = request.body.read
+    request.body.rewind
+    begin
+      parsed_body = JSON.parse(body)
+      mcp_logger.info("Request: #{parsed_body["method"]} (id: #{parsed_body["id"]})")
+      mcp_logger.debug("Request body: #{JSON.pretty_generate(parsed_body)}")
+    rescue JSON::ParserError
+      mcp_logger.warn("Request body (raw): #{body}")
+    end
+  end
+
+  # Handle the request
+  response = transport.handle_request(request)
+
+  # Log the MCP response details
+  _, _, body = response
+  if body.is_a?(Array) && !body.empty?
```

Comment:
```suggestion
  if body.is_a?(Array) && !body.empty? && body.first
```

I was still seeing some errors when testing this in the MCP Inspector 


```
MCP] Request body: {
  "jsonrpc": "2.0",
  "method": "notifications/initialized"
}
TypeError: no implicit conversion of nil into String (TypeError)

    Parser.parse(source, opts)
                 ^^^^^^^^^^^^
	/Users/topherbullock/.gem/ruby/3.3.0/gems/json-2.12.0/lib/json/common.rb:338:in `parse'
	/Users/topherbullock/.gem/ruby/3.3.0/gems/json-2.12.0/lib/json/common.rb:338:in `parse'
	./examples/http_server.rb:130:in `block in <main>'
	/Users/topherbullock/.gem/ruby/3.3.0/gems/rack-3.1.12/lib/rack/show_exceptions.rb:31:in `call'
	/Users/topherbullock/.gem/ruby/3.3.0/gems/rack-3.1.12/lib/rack/common_logger.rb:43:in `call'
	/Users/topherbullock/.gem/ruby/3.3.0/gems/rack-3.1.12/lib/rack/builder.rb:277:in `call'
```